libsignal-protocol/src/crypto/openssl.rs

```diff
@@ -68,14 +70,18 @@ impl Crypto for OpenSSLCrypto {
 
     fn hmac_sha256(
         &self,
-        _key: &[u8],
+        key: &[u8],
     ) -> Result<Box<dyn Sha256Hmac>, InternalError> {
-        let nid = Nid::HMACWITHSHA256;
-        let ty = MessageDigest::from_nid(nid)
-            .ok_or_else(|| InternalError::Unknown)?;
-        let hasher = Hasher::new(ty).map_err(|_e| InternalError::Unknown)?;
-
-        Ok(Box::new(hasher))
+        let pkey =
+            Box::new(PKey::hmac(key).map_err(|_e| InternalError::Unknown)?);
```

Comment:
yup, it's a leak
![image](https://user-images.githubusercontent.com/14620076/59145026-9827c900-89de-11e9-98cd-ce60711a0e67.png)

Reply:
solved using `rental` crate